FirebaseAdmin/FirebaseAdmin.Tests/Messaging/FirebaseMessagingTest.cs

```diff
@@ -54,6 +54,17 @@ public void GetMessaging()
             Assert.Throws<InvalidOperationException>(() => FirebaseMessaging.GetMessaging(app));
         }
 
+        [Fact]
+        public void GetMessagingWithClientFactory()
+        {
+            var app = FirebaseApp.Create(new AppOptions() { Credential = MockCredential, HttpClientFactory = new HttpClientFactory() }, "MyApp");
+            FirebaseMessaging messaging = FirebaseMessaging.GetMessaging(app);
+            Assert.NotNull(messaging);
+            Assert.Same(messaging, FirebaseMessaging.GetMessaging(app));
+            app.Delete();
+            Assert.Throws<InvalidOperationException>(() => FirebaseMessaging.GetMessaging(app));
```

Comment:
We should try to do a stronger check here. Otherwise a future code change can silently break this functionality. Easiest option would be to inject a `MockHttpClientFactory` via `AppOptions`. Then invoke `FirebaseMessaging.SendAsync()` and see if that factory get called.